lib/diagnostic.js

```diff
@@ -6,7 +6,13 @@
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
 const min = function (previous, current) {
-
+  let minimum = null
+  previous.reduce(function (minimum) {
```

Comment:
Nice! Although you didn't have to write out this function as a `reduce`. The wording for this question might be a little unclear, but you were only required to provide the function that we would pass to a `reduce` function.